src/TConsole.cpp

```diff
@@ -51,6 +51,7 @@
 using namespace std;
 
 const QString TConsole::cmLuaLineVariable("line");
+const QString cMainConsoleName(QStringLiteral("main"));
```

Comment:
Because this will be used throughout the whole class file - and avoids the penalty of multiple un-share-able *identical* copies of a separate instance for *each* usage in the file.

Reply:
I don't think that penalty of repeated 4 characters is worth this (ugly) abstraction. How many bytes is it actually saving in the zipped up final release?

Reply:
I think I had counted around 50 instances in the TLuaInterpreter file - and I don't think it is ugly - but instead a reasonable (and a lot better then a preprocessor directive BTW) way of centralising a value that is used extensively - I would consider a shorter name for the constant if you have a better one. And - should it *ever* be needed/desired to translate it for a different locale it would at least be do-able...

Reply:
It's used in scripts, so there's no way we'll change it, and the code looks uglier and more difficult to write imho. @keneanung what do you reckon?

Reply:
I agree with both of you to a part:
@vadi2 is right in that this abstraction is not enough if value to me to have such a complicated construct here and on every call site.
But I also agree with @SlySven that a constant would be useful. Not for renaming the main console, but for having IDE and compiler support to avoid typos in the constant.

Can't we simply use a standard C/C++ string? The constant is ASCII only and we convert it to a a standard string anyways.

Additionally, if we decide for a constant, can we have one in the whole project and not define it in every class again?

Reply:
Typos in the constant is not an issue that we've ever had - this would be solving an non-existent problem to me.

Reply:
> Can't we simply use a standard C/C++ string? The constant is ASCII only and we convert it to a a standard string anyways.

Actually as the string is/will be used as a `QString` when it gets to the main `TConsole` where it will be looked up in the `(QMap<QString, TConsole*>) TConsole::mSubConsoleMap` it seems pointless to avoid not having it as one to start with. Some `TConsole` function do/did `std::string` arguments but then go on to convert them to `QString`s so it seemed simpler to use the latter from end to end (and pass them by constant reference, though that is moot as I understand it as Qt's COW in conjunction with passing `const` *values* will optimise out the copying of function arguments won't it?)

> Additionally, if we decide for a constant, can we have one in the whole project and not define it in every class again?

Would be nice but then where does it get defined so as to avoid making it even longer: `TConsole::cMainConsoleName` ???

Reply:
I'd be happier if we called it `main`. Because `cMainConsoleName` at 17 characters is worse than `QStringLiteral("main")` which is 23 but understandable right away, it makes no sense to me to use a confusing constant which isn't that much shorter than the real thing.